basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,7 +234,208 @@ namespace basic_autonomy
             return constrained_points;
         }
 
-        std::vector<PointSpeedPair> create_lanechange_geometry(const cav_msgs::Maneuver &maneuver, double starting_downtrack,
+        std::vector<lanelet::BasicPoint2d> create_lanechange_geometry(lanelet::Id starting_lane_id, lanelet::Id ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state)
+        {
+            std::vector<lanelet::BasicPoint2d> centerline_points;
+
+            //Get starting lanelet and ending lanelets
+            lanelet::ConstLanelet starting_lanelet = wm->getMap()->laneletLayer.get(starting_lane_id);
+            lanelet::ConstLanelet ending_lanelet = wm->getMap()->laneletLayer.get(ending_lane_id);
+            
+            lanelet::ConstLanelets starting_lane;
+            starting_lane.push_back(starting_lanelet);
+
+            std::vector<lanelet::BasicPoint2d> reference_centerline;
+            // 400 value here is an arbitrary attempt at improving performance by reducing copy operations. 
+            // Value picked based on annecdotal evidence from STOL system testing
+            reference_centerline.reserve(400);
+            bool shared_boundary_found = false;
+            bool is_lanechange_left = false;
+            
+            lanelet::BasicLineString2d current_lanelet_centerline = starting_lanelet.centerline2d().basicLineString();
+            lanelet::ConstLanelet current_lanelet = starting_lanelet;
+            reference_centerline.insert(reference_centerline.end(), current_lanelet_centerline.begin(), current_lanelet_centerline.end());
+
+            while(!shared_boundary_found){
+                //Assumption here is for lane change to happen between two adjacent lanelets, they must share a lane boundary (linestring)
```

Comment:
Good to list the assumption here, but its really needed at the interface comment rather than in the implementation. Assumptions need to be understood by the method user.